sdk/communication/azure-communication-administration/azure/communication/administration/_phone_number_administration_client.py

```diff
@@ -458,19 +522,50 @@ def cancel_search(
         )
 
     @distributed_trace
-    def purchase_search(
-            self,
-            search_id,  # type: str
-            **kwargs  # type: Any
+    def begin_purchase_reservation(
+        self,
+        **kwargs  # type: Any
     ):
-        # type: (...) -> None
-        """Purchases the phone number search.
-
-        :param search_id: The search id to be purchased.
-        :type search_id: str
-        :rtype: None
+        # type: (...) -> LROPoller
+        """Begins purchase the reserved phone numbers of a phone number search.
+        Caller must provide either search_id, or continuation_token keywords to use the method.
+        If both body and continuation_token are specified, only continuation_token will be used to
+        restart a poller from a saved state, and keyword search_id will be ignored.
+        :keyword str search_id: The search id to be purchased.
+        :keyword str continuation_token: A continuation token to restart a poller from a saved state.
+        :rtype: ~azure.core.polling.LROPoller[~azure.communication.administration.PhoneNumberSearch]
         """
-        return self._phone_number_administration_client.phone_number_administration.purchase_search(
-            search_id,
+        cont_token = kwargs.pop('continuation_token', None)  # type: Optional[str]
+
+        search_polling = PurchaseReservationPolling(
+            is_terminated=lambda status: status in [
+                SearchStatus.Success,
+                SearchStatus.Expired,
+                SearchStatus.Cancelled,
+                SearchStatus.Error
+            ]
+        )
+
+        if cont_token is not None:
+            return LROPoller.from_continuation_token(
+                polling_method=search_polling,
+                continuation_token=cont_token,
+                client=self._phone_number_administration_client.phone_number_administration
+            )
+
+        if "search_id" not in kwargs:
```

Comment:
Do we want to rename search_id -> reservation_id everywhere?

Reply:
Sure. Fixing.